### PR TITLE
Create landing page and add Google Sign Option

### PR DESCRIPTION
**Pull request by Maryam676**
- Created the initial landing page for user
- One sign in option for now 
- Appengine and firebase dependencies added to pom file
- Firebase app manager class for ensuring app is only created once

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SessionInit.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import com.google.gson.Gson;
+
+@WebServlet("/init")
+public class SessionInit extends HttpServlet {
```
Comment (nmittu) [inline src/main/java/com/google/sps/servlets/SessionInit.java]: Why is this servlet empty?

File: src/main/java/com/google/sps/servlets/VerifySession.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/session-verify")
+public class VerifySession extends HttpServlet {
+
+  // Obtain the cookie from the current session
+  static String getCookie(HttpServletRequest request, String name, String defaultValue) {
```
Comment (nmittu) [inline src/main/java/com/google/sps/servlets/VerifySession.java]: Move this to a helper class. Something like `RequestUtils`. This method can also be made `public`.

File: src/main/java/com/google/sps/servlets/VerifySession.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import javax.servlet.http.Cookie;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/session-verify")
+public class VerifySession extends HttpServlet {
```
Comment (nmittu) [inline src/main/java/com/google/sps/servlets/VerifySession.java]: Try not to merge in incomplete servlets.

**Review by nmittu — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/SessionInit.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
```
Comment (nmittu) [inline src/main/java/com/google/sps/servlets/SessionInit.java]: Using a wildcard in imports can cause issues if the package contains a Class that interferes with other classes your using.  

https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

**Review by nmittu — CHANGES_REQUESTED** [review]

**Review by nmittu — APPROVED** [review]

**Review by biancasap — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/VerifySession.java [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (mdchu) [inline src/main/java/com/google/sps/servlets/VerifySession.java:1]: stray line edit in otherwise unchanged file?

**Review by mdchu — APPROVED** [review]

**Pull request merged**